cabalscan/src/CabalScan/RuleGenerator.hs

```diff
@@ -293,16 +293,15 @@ depPackageNames = concatMap depNames . Cabal.targetBuildDepends
 --
 -- The directories in @hsSourceDirs@ must be relative to @parentDir@.
 findModulePaths :: Path b Dir -> [Path Rel Dir] -> Path Rel File -> IO [Path Rel File]
-findModulePaths parentDir hsSourceDirs modPath =
+findModulePaths parentDir hsSourceDirs modulePath =
   case hsSourceDirs of
     [] -> return []
     srcDir:otherDirs -> do
-      modulePath <- Path.parseRelFile (Path.toFilePath modPath)
```

Comment:
Is there some reason to do this? If the internal invariant of `Path` is broken and that's why we're doing this, then we should fix that instead by not using unsafe `Path` builders.

Reply:
I don't think this is intended. Might be an accident after some evolution. Thanks for pointing it.